pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java

```diff
@@ -82,6 +90,10 @@ private void onLatestBlockHead(final Block headBlock) {
     if (headAtFollowDistance
         .map(current -> current.compareTo(newHeadAtFollowDistance) < 0)
         .orElse(true)) {
+      if (!reachedHead.get()) {
```

Comment:
```suggestion
      if (reachedHead.compareAndSet(false, true)) {
```

Then drop the `reachedHead.set(true)`.  That way you're using a single atomic operation to check and update so it's thread safe.